modules/yarpc/middleware.go

```diff
@@ -109,36 +98,31 @@ func authorize(ctx context.Context, host service.Host, statsClient *statsClient)
 	return ctx, nil
 }
 
-type panicInboundMiddleware struct {
-	statsClient *statsClient
-}
+type panicInboundMiddleware struct{}
 
 func (p panicInboundMiddleware) Handle(
 	ctx context.Context,
 	req *transport.Request,
 	resw transport.ResponseWriter,
 	handler transport.UnaryHandler,
 ) error {
-	defer panicRecovery(ctx, p.statsClient)
+	defer panicRecovery(ctx)
 	return handler.Handle(ctx, req, resw)
 }
 
-type panicOnewayInboundMiddleware struct {
-	statsClient *statsClient
-}
+type panicOnewayInboundMiddleware struct{}
 
 func (p panicOnewayInboundMiddleware) HandleOneway(
 	ctx context.Context,
 	req *transport.Request,
 	handler transport.OnewayHandler,
 ) error {
-	defer panicRecovery(ctx, p.statsClient)
+	defer panicRecovery(ctx)
 	return handler.HandleOneway(ctx, req)
 }
 
-func panicRecovery(ctx context.Context, statsClient *statsClient) {
+func panicRecovery(ctx context.Context) {
 	if err := recover(); err != nil {
-		statsClient.RPCPanicCounter().Inc(1)
```

Comment:
Does YARPC already have it? In order to check panics users will have to set alerts for log messages, instead of metrics, not sure if it is always possible.